atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CandidateCellForSweeping.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.api;
+
+import javax.annotation.Nullable;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface CandidateCellForSweeping {
+
+    Cell cell();
+
+    /**
+     * All start timestamps for the cell that are strictly less than
+     * {@link CandidateCellForSweepingRequest#sweepTimestamp()} and are not in
+     * {@link CandidateCellForSweepingRequest#timestampsToIgnore()}, in ascending order.
+     *
+     * Note that this will be null if and only if {@link #isCandidate()} is false.
+     */
+    @Nullable
+    long[] sortedTimestamps();
+
+    /**
+     * If {@link CandidateCellForSweepingRequest#shouldCheckIfLatestValueIsEmpty()} was set to true,
+     * then this method returns true if and only if the value corresponding to the last entry of
+     * {@link #sortedTimestamps()} is empty.
+     * <p>
+     * Otherwise, the return value is undefined and depends on the implementation.
+     */
+    boolean isLatestValueEmpty();
```

Comment:
question about this, what is the behavior if the last value in the list of timestamps here is empty but there was a cell at a later timestamp (later than the sweeping request) that was non-empty? does this matter? (and what about thorough sweep?)  I'm not sure if these make a difference, but might be nice to have somewhere that documents how sweep is supposed to work in these cases.